src/styles/buttons.scss

```diff
@@ -4,124 +4,215 @@
+
+  &::after {
+    content: "";
+    display: block;
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    pointer-events: none;
+    background-image: radial-gradient(circle, $ripple-color 10%, transparent 10.01%);
+    background-repeat: no-repeat;
+    background-position: 50%;
+    mix-blend-mode: $blend-mode;
+    transform: scale(10, 10);
+    opacity: 0;
+    transition: transform 0.3s, opacity 0.8s;
+  }
+
+  &:active {
+    &::after {
+      transform: scale(0, 0);
+      opacity: $opacity;
+      transition: 0s;
+    }
+  }
+}
+
 .btn {
-  @include font('body');
+  box-sizing: border-box;
   display: inline-flex;
-  text-decoration: none;
-  border: none;
-  border-radius: $default-border-radius;
-  background-color: transparent;
-  padding: 13px $stack-m;
-  justify-content: center;
-  width: auto;
   align-items: center;
+  justify-content: center;
+  margin: 0;
+  padding: $stack-s $inline-l;
+  border: 0;
+  border-radius: 2px;
+  outline: 0;
+  background-color: transparent;
+  font-family: $font-primary;
   font-weight: 600;
+  font-size: 16px;
+  line-height: 28px;
   text-align: center;
+  text-decoration: none;
+  -webkit-appearance: none;
+  -moz-appearance: none;
+
   &:hover {
     cursor: pointer;
   }
 }
 
-.btn--primary,
-.btn--secondary {
+.btn--primary {
+  @include btn-ripple;
+
   background-color: getShade($secondary-color, 110);
+  box-shadow: 0px 3px 2px rgba(0, 0, 0, 0.12), 0px 0px 4px rgba(0, 0, 0, 0.24);
   color: $white;
-  font-family: $font-primary;
-  width: 100%;
-  font-size: 16px;
-  height: 52px;
-  padding: 0 $stack-m;
-  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.3);
-  transition: $button-transition;
+  transition: 
+    background-color 0.3s ease-in-out,
+    box-shadow 0.3s ease-in-out,
+    color 0.3s ease-in-out;
+
   &:hover {
-    color: $white;
-    box-shadow: 0 3px 2px 0 rgba(0,0,0,0.3);
-    background-color: getShade($secondary-color, 110);
-    transform: translateY(-1px);
+    box-shadow: 0px 6px 12px rgba(0, 0, 0, 0.3);
   }
-  
-  // Ripple effect on click
-  position: relative;
-  overflow: hidden;
-  transform: translate3d(0,0,0);
-  &:after {
-    content: "";
-    display: block;
-    position: absolute;
-    width: 100%;
-    height: 100%;
-    top: 0;
-    left: 0;
-    pointer-events: none;
-    background-image: radial-gradient(circle,#000 10%,transparent 10.01%);
-    background-repeat: no-repeat;
-    background-position: 50%;
-    transform: scale(10,10);
-    opacity: 0;
-    transition: transform 300ms, opacity 0.8s;
 
+  &:focus {
+    box-shadow: 0px 0px 14px getShade($secondary-color, 110), 0px 0px 1px getShade($secondary-color, 110);
   }
-  &:active:after {
-    transform: scale(0,0);
-    opacity: .2;
-    transition: 0s;
+
+  &:active {
+    background-color: getShade($secondary-color, 110);
   }
 
-  @media (min-width: $break-tablet) {
-    font-size: 18px;
-    width: auto;
+  &[disabled],
+  &.btn--primary__disabled {
```

Comment:
Do we actually need this `.btn--primary__disabled` (and for the other modifiers as well)?
I think the attribute check should be enough here.
In can we need also a class definition, I would suggest using `btn--disabled` (in combination with the regular modifier), that would make it BEM compliant.

Reply:
Will update it to be `.btn--disabled` :D!